src/qute/commands/registerCommands.ts

```diff
@@ -214,6 +227,42 @@ function getSettingsValue(value: any, section: string, editType: ConfigurationIt
   return value;
 }
 
+const LANGUAGE_MAP = new Map<string, string>([
+  [".html", "qute-html"],
+  [".htm", "qute-html"],
+  [".txt", "qute-txt"],
+  [".yaml", "qute-yaml"],
+  [".yml", "qute-yaml"],
+  [".json", "qute-json"]
+]);
+
+/**
+ * Update the language ID to a supported Qute language id if needed.
+ *
+ * @param context the extension context
+ * @param document the text document.
+ * @param onExtensionLoad if the user manually changed the language id.
+ */
+ async function updateQuteLanguageId(context: ExtensionContext, document: TextDocument, onExtensionLoad: boolean) {
+  const propertiesLanguageMismatch: QuteTemplateLanguageMismatch = QuteSettings.getQuteTemplatesLanguageMismatch();
+  // Check if the setting is set to ignore or if the language ID is already set to Qute
+  if (propertiesLanguageMismatch === QuteTemplateLanguageMismatch.ignore || document.languageId.startsWith('qute-')) {
+    // Do nothing
+    return;
+  }
+  const fileName: string = path.basename(document.fileName);
+  if (document.fileName.includes(`resources${path.sep}templates${path.sep}`)) {
```

Comment:
When a HTML file comes from JAR, the document.fileName looks like this `\\templates\\tags\\ifError.html`, it should be really nice if you could support that too.

Reply:
If you're referring to a supported file extension that is nested in a directory within `resources/templates`, this is already supported:
![nestedtemplatelanguagesupport](https://user-images.githubusercontent.com/26389510/154117929-2ae2aa13-5d24-48dc-be74-a930839d5c2e.gif)

Reply:
No I speak about html file hosted in a JAR. Please ignore my comment we will fix that in an another PR